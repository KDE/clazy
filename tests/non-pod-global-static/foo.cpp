#include <QtCore/QCoreApplication>
#include <QtCore/QtPlugin>






struct NonPod
{
    NonPod() {}
    ~NonPod() {}
    int a;
};

struct NonPod2
{
    NonPod2() {}
};

struct NonPod3
{
    ~NonPod3() {}
};


struct Pod
{
    int a;
};


static Pod p;
static NonPod p1;
static NonPod2 p2;
static NonPod3 p3;
static NonPod* p4;
static int p5;

void foo() {}
Q_COREAPP_STARTUP_FUNCTION(foo) // OK, it's blacklisted

struct NonPodButOk
{
    NonPodButOk() = default;
    NonPodButOk(const NonPodButOk &) {}
};

static NonPodButOk p6; // OK, both called ctor and dtor are trivial

NonPod p7; // OK, it's not static, might be used somewhere else

struct HasConstExprCtor
{
    constexpr HasConstExprCtor(int i) : m_i(i) {}
    const int m_i;
};

constexpr static HasConstExprCtor p8(1);
static HasConstExprCtor p9(1);

namespace a_name_space{
  static NonPod NonPod8;
}


Q_IMPORT_PLUGIN(QWindowsIntegrationPlugin) // OK, it's blacklisted
