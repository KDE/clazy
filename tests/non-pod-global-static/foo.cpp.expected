non-pod-global-static/foo.cpp:34:1: warning: non-POD static (NonPod p1) [-Wclazy-non-pod-global-static]
non-pod-global-static/foo.cpp:35:1: warning: non-POD static (NonPod2 p2) [-Wclazy-non-pod-global-static]
non-pod-global-static/foo.cpp:36:1: warning: non-POD static (NonPod3 p3) [-Wclazy-non-pod-global-static]
non-pod-global-static/foo.cpp:63:3: warning: non-POD static (NonPod a_name_space::NonPod8) [-Wclazy-non-pod-global-static]
